### DISCUSSION TITLE
Why is Axios a dependency?

### DISCUSSION THREAD

**Discussion by alavkx** [ecyrbe/zodios#169] [Q&A] [answered]
Why the dependency on `axios`? Are there some features that were easier to add that can't be achieved with `fetch`? No shade just curious if there is an opportunity to thin the abstraction and trim ~10kb.

**Comment by alavkx** [comment]
I've just now seen your note 😅 

> It's worth noting, that you should not use the fetch plugin on nodejs. Indeed, fetch lacks a lot of features on backend side and you should use axios default http adapter for node (default). If you still want to use fetch on your backend, you should use a polyfill, zodios does not provide one.
> Do not open an issue for fetch support on nodejs unless you are willing to add support for it with a PR at the same time. I might reconsider this position in the future when fetch becomes feature complete on nodejs.

Perfectly reasonable 👍

**Comment by ecyrbe** [comment, marked as answer]
Hello,

Thank you for taking the time to read the documentation.
First of all, i want to emphasize, that zodios was build to answer my professional needs first.
So, Axios is a dependancy because i rely on axios features and did not want to reinvent them.
Axios does so much more than fetch. adding the same functionnalities than axios to fetch will usually mean installing other dependancies that will surpass the 10k you'll save from using just fetch.
The only argument that matters in my point of view is not about those 10k, but about not having the fetch options available to you when using axios.  
That's why i created the fetch plugin. This way you can still use Zodios but with fetch behind the scene to have control over :  
- keepalive
- cors
- referer policy

**Reply by alavkx** [reply]
Thank you @ecyrbe! ❤️
